src/lightning_lite/accelerators/cuda.py

```diff
@@ -77,6 +78,25 @@ def _get_all_available_cuda_gpus() -> List[int]:
     return list(range(num_cuda_devices()))
 
 
+@contextmanager
+def _patch_cuda_is_available() -> Generator:
+    """Context manager that safely patches :func:`torch.cuda.is_available` with its NVML-based version if
+    possible."""
+    orig_check = None
+    new_check = torch.cuda.device_count if _TORCH_GREATER_EQUAL_1_13 else _device_count_nvml
```

Comment:
Can you make it just exit if >=1.13? No patching at all

Reply:
In PyTorch 1.13 as it stands, `torch.cuda.is_available()` still uses a CUDA Runtime API based availability assessment and therefore poisons subsequent forks. I have a [PR open](https://github.com/pytorch/pytorch/pull/85951) (that @awaelchli has been awesome reviewing, thanks man!) in PyTorch currently that may allow us to request NVML-based behavior for `torch.cuda.is_available()` throughout Lightning without needing to patch the function or add Lightning's own NVML-based assessment implementation. I don't know yet if that behavioral enhancement will make it into 1.13 though so I think we need to assume continued patching with 1.13 for now.

Reply:
Exactly! I think we need to first know if it really goes in to 1.13. We can open an issue on Lighting as a reminder that we want to evaluate this again when 1.13 is out.